packages/story-editor/src/app/canvas/utils/getLayerArrangementProps.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { LAYER_DIRECTIONS } from '../../../constants';
+
+function getLayerArrangementProps(key, shift, selectedElements, elements) {
+  // This only supports moving single layer.
+  if (!selectedElements || selectedElements.length > 1) {
+    return {};
+  }
+  let position = null;
+  if (key === 'ArrowUp') {
+    position = shift ? LAYER_DIRECTIONS.FRONT : LAYER_DIRECTIONS.FORWARD;
+  }
+  if (key === 'ArrowDown') {
+    position = shift ? LAYER_DIRECTIONS.BACK : LAYER_DIRECTIONS.BACKWARD;
+  }
+  if (!position) {
+    return {};
+  }
+
+  const { id, groupId } = selectedElements[0];
+  // Get layer index.
+  const currentPosition = elements.findIndex(({ id: elId }) => elId === id);
+
+  // If the layer is in a group, check if it's first or last of the group,
+  // in this case we should just move it out of the group.
+  if (groupId) {
+    const isLastInGroup = elements[currentPosition - 1]?.groupId !== groupId;
+    const isFirstInGroup = elements[currentPosition + 1]?.groupId !== groupId;
+    if (
+      (isLastInGroup && position === LAYER_DIRECTIONS.BACKWARD) ||
+      (isFirstInGroup && position === LAYER_DIRECTIONS.FORWARD)
+    ) {
+      return {
+        position: currentPosition,
+        groupId: null,
+      };
+    }
```

Comment:
Note here, that if the element is both first and last and will be moved out of the group, the group should also be removed, which currently doesn't happen. If you move the last element out of a group, the group object persists in the groups object on the page, but it cannot be used anymore because no element references it.

We could add this to a follow-up, but I think we could just add a `removeGroup: isFirstInGroup && isLastInGroup` and then in `useCanvasKeys` check for that prop. Or we could update the underlying `arrangeElement` reducer to always check if a group is left empty and then delete it - that would probably be the proper way to do it actually.

Reply:
Updated the reducer in [9c1f855](https://github.com/GoogleForCreators/web-stories-wp/pull/11884/commits/9c1f8555b5e9d484f5efc94af63799d413c5f86c), looks like generally the groups were not being cleaned up elsewhere either so didn't see any other code that needed adjusting. Let me know if I missed any place.